src/platforms/web/compiler/directives/model.js

```diff
@@ -40,7 +40,7 @@ export default function model (
     }
   }
 
-  if (el.component) {
+  if (el.component && (!el.attrsMap.is || !isInputTag(el.attrsMap.is))) {
```

Comment:
I wonder if we shouldn't use ` && !isInputTag(el.attrsMap.is)` and call `makeMap` with `false` (or nothing). If someone uses an capitalized name for the component, like `Textarea` or `Input`, they probably want to use a component instead of the HTML native tag